zulipterminal/ui_tools/buttons.py

```diff
@@ -475,6 +499,13 @@ def _switch_narrow_to(self, parsed_link: ParsedNarrowLink) -> None:
             self.topic_name = parsed_link['topic_name']
             self.message = dict(id=parsed_link['message_id'])
             self.controller.narrow_to_topic(self)
+        elif 'pm-with' == narrow:
```

Comment:
What I meant about my last comment is that I'm fairly sure there are valid pm-with and pm-with:near (equivalent in URLs), so we need to likely handle both cases.